### PR TITLE
[DanaKit] Progress update 02-06-2024

### PR DESCRIPTION
**Pull request by bastiaanv**
A long overdue update for the DanaKit project. This PR contains (but not limited to) the following changes:
- fix for Loop cycle hangs
- Lower limit of time difference (was 1 minute, but caused double insulin entries)
- Rename all the hardcoded Loop to the dynamic appName (thanks @marionbarker!)
- Added error messages for failed bolus (daily limit violation or bolus timeout error, i.e.)
- Improved static basal steps (since a Dana pump supports basal steps of 0.01U/hr)
- Optimize connection logic/sync (some data will only be synced during the sync function instead of at every connection)

And a sync with dev :)

### CODE REVIEW

File: DanaKit
Comment (Sjoerd-Bo3) [inline DanaKit:1]: CleanShot 2024-06-02 at 17 50 29@2x
CleanShot 2024-06-02 at 17 50 14@2x

Apart from the above Typo's (vv1), LGTM!

**Review by Sjoerd-Bo3 — CHANGES_REQUESTED** [review]
Apart from the Typo's (vv1), LGTM!

**Review by Sjoerd-Bo3 — APPROVED** [review]

**Review by marionbarker — APPROVED** [review]
### Tested:
* iPhone SE running 17.5.1, Trio dev (commit 14c76432)
* connected to rPI DASH simulator
* Built feat/dana (commit 13dfdec2)
* The pod behaved nominally.
* disconnect rPi pod
* connect rPi dana-simulator
* did a bolus but then ran into simulator issues (the rPi never finishes the bolus)

Assume the issues I am seeing are due to the rPi simulator.
Is there anyone with a real Dana pump who can test this?

**Comment by marionbarker** [comment]
I updated to most recent commit and configured to the new version of the dana-simulator.
* see [Miscellaneous issues with dana-simulator](https://github.com/bastiaanv/dana-simulator/issues/2#issuecomment-2153411945)

### What works
* I can now bolus more than once (simulator update)
* I can get a green loop whether glucose is rising or failling

### What does not work
I can't figure out how to delete Dana once it is selected.

I think when I delete the Dana pump for the Dana menu - it should indicate no pump on main screen.
I can't figure out how to switch away from Dana once it has been selected.
* main screen shows !! No insulin
   * tapping on that icon takes you to choose which dana pump you want
* Settings -> pump also takes you to choose which dana pump you want
* You can scan and it finds it (because rPi is still connected).

Next try to delete Dana pump and control C out of simulator.
Quit the Trio app and rebuild.
It still only allows user to reconnect to Dana pump.
Restarting the rPi simulator allows Trio to connect.

@avouspierre Can you help on the deleting pump part?

**Comment by bjornoleh** [comment]
Merging with two approvals

**Pull request merged**